templates/hooks--fsmonitor-watchman.sample

```diff
@@ -22,7 +22,8 @@ my ($version, $time) = @ARGV;
 
 if ($version == 1) {
 	# convert nanoseconds to seconds
-	$time = int $time / 1000000000;
+	# substract one second to make sure watchman will return all changes
```

Comment:
Do you know if the "one second" issue is Mac specific?

Reply:
Related you mentioned:

> watchman returns are not inclusive of the time that is passed to it

How did you track this down?  If you found this in the Watchman source could you provide a link to it?

Reply:
(not the author, but answering because I've been following along and want to double-check my understanding)

We need to back off a second because we're snapping these queries to the second boundary, and the `since` expression in watchman performs a greater-than check (https://facebook.github.io/watchman/docs/expr/since.html). If we don't back off one unit of what we're querying by, and in this case it is seconds, x>x is always false.

@kewillford @derrickstolee did I get that correct?

Reply:
Yes and the documentation on [clockspec](https://facebook.github.io/watchman/docs/clockspec.html) says "Using a timestamp is prone to race conditions in understanding the complete state of the file tree."

Reply:
Is it possible to use the abstract clockid to help guard against these race conditions?

Reply:
@jrbriggs: @kewillford and I talked about using the clock time string that Watchman provides, but it requires adding an extension to the index. Definitely something to look into, but not appropriate for this fix.